testsuite/pytests/sli2py_synapses/test_inh_stdp_synapse.py

```diff
@@ -0,0 +1,221 @@
+    pg_pre = nest.Create("poisson_generator", params={"rate": nu})
+
+    parrot = nest.Create("parrot_neuron")
+    neuron = nest.Create("iaf_psc_alpha", params={"tau_minus": tau_minus})
+
+    sr_pre = nest.Create("spike_recorder")
+    sr_post = nest.Create("spike_recorder")
+
+    # Set defaults for stdp_synapse
+    # Note: "lambda" is a Python keyword, but can be used as a dict key
+    nest.SetDefaults(
+        "stdp_synapse",
+        {
+            "weight": w_neuron,
+            "alpha": alpha,
+            "lambda": lambda_val,  # noqa: A001
+            "tau_plus": tau_plus,
+            "mu_plus": mu_plus,
+            "mu_minus": mu_minus,
+            "Wmax": w_max,
+        },
+    )
+
+    # Connect
+    nest.Connect(pg_exc, neuron, syn_spec={"weight": w_exc, "delay": delay})
+    nest.Connect(pg_inh, neuron, syn_spec={"weight": w_inh, "delay": delay})
+    nest.Connect(pg_pre, parrot, syn_spec={"weight": w_exc, "delay": delay})
+
+    nest.Connect(parrot, neuron, syn_spec={"synapse_model": "stdp_synapse", "weight": w_neuron, "delay": delay})
+
+    nest.Connect(parrot, sr_pre)
+    nest.Connect(neuron, sr_post)
+
+    # Simulate and get data
+    nest.Simulate(10000.0)
+
+    pre_spikes = np.array(sr_pre.get("events", "times")) + axonal_delay
+    post_spikes = np.array(sr_post.get("events", "times")) + backpr_delay
+
+    # Get final weight
+    conns = nest.GetConnections(source=parrot, target=neuron)
+    assert len(conns) == 1, "Should have exactly one connection"
+    final_weight = conns[0].get("weight")
+
+    # Check final weight by reproducing STDP calculations
+    K_plus = 0.0
+    K_minus = 0.0
+    last_pre = 0.0
+    last_post = 0.0
+    j = 0
+    i = 0
+
+    if len(post_spikes) == 0 or len(pre_spikes) == 0:
+        pytest.skip("No spikes recorded, cannot test STDP")
+
+    post_spike = post_spikes[i] if i < len(post_spikes) else float("inf")
+    pre_spike = pre_spikes[j] if j < len(pre_spikes) else float("inf")
+    w = w_neuron / w_max  # normalized weight
+
+    def update_K_plus():
+        nonlocal K_plus
+        K_plus = K_plus * np.exp((last_pre - pre_spike) / tau_plus) + 1.0
+
+    def update_K_minus():
+        nonlocal K_minus
+        K_minus = K_minus * np.exp((last_post - post_spike) / tau_minus) + 1.0
+
+    def next_pre_spike():
+        nonlocal j, last_pre, pre_spike
+        j += 1
+        last_pre = pre_spike
+        if j < len(pre_spikes):
+            pre_spike = pre_spikes[j]
+        else:
+            pre_spike = float("inf")
+
+    def next_post_spike():
+        nonlocal i, last_post, post_spike
+        i += 1
+        last_post = post_spike
+        if i < len(post_spikes):
+            post_spike = post_spikes[i]
+        else:
+            post_spike = float("inf")
+
+    def facilitate():
+        nonlocal w
+        w = w + lambda_val * ((1.0 - w) ** mu_plus) * K_plus * np.exp((last_pre - post_spike) / tau_plus)
```

Comment:
Brackets not required, but fine if kept for clarity:
```suggestion
        w = w + lambda_val * (1.0 - w) ** mu_plus * K_plus * np.exp((last_pre - post_spike) / tau_plus)
```